code/modules/surgery/organs/cybernetics/augments_internal_wordblock.dm

```diff
@@ -0,0 +1,164 @@
+GLOBAL_LIST_EMPTY(wordblockers)
+
+/obj/item/organ/cyberimp/brain/wordblocker
+	name = "speech filtration implant"
+	desc = "Originally made for prison use, this gruesome implant interfaces directly with the brain's speech centers, both allowing control over what the victim can say and bypassing need for a cyberlink."
+	icon_state = "brain_implant"
+	slot = ORGAN_SLOT_BRAIN_LANGUAGEINHIBITOR
+	encode_info = AUGMENT_NO_REQ
+
+	var/id = 0
+	var/static/wordblock_uid = 0
+	var/locked = FALSE // write protection (for idk admin shenanigans)
+
+	var/datum/wordfilter_manager/mgr = null
+
+/obj/item/organ/cyberimp/brain/wordblocker/New()
+	id = wordblock_uid + 1
+	wordblock_uid = id
+	mgr = new(id, src)
+
+	GLOB.wordblockers.Add(src)
+	..()
+
+/obj/item/organ/cyberimp/brain/wordblocker/Insert(mob/living/carbon/owner, special)
+	..()
+	RegisterSignal(owner, COMSIG_MOB_SAY, .proc/handle_speech)
+
+/obj/item/organ/cyberimp/brain/wordblocker/Remove(mob/living/carbon/organ_owner, special)
+	UnregisterSignal(owner, COMSIG_MOB_SAY)
+	..()
+
+/obj/item/organ/cyberimp/brain/wordblocker/ComponentInitialize()
+	. = ..()
+	// We do not want our encode_info scrambled
+	AddElement(/datum/element/empprotection, EMP_PROTECT_SELF)
+
+/obj/item/organ/cyberimp/brain/wordblocker/Del()
+	GLOB.wordblockers.Remove(src)
+	UnregisterSignal(owner, COMSIG_MOB_SAY)
+	..()
+
+/obj/item/organ/cyberimp/brain/wordblocker/proc/get_manager()
+	if(!mgr)
+		mgr = new(id, src)
+	return mgr
+
+/obj/item/organ/cyberimp/brain/wordblocker/proc/handle_speech(datum/source, list/speech_args)
+	SIGNAL_HANDLER
+
+	if(!mgr)
+		return
+
+	var/newmsg = mgr.process_message(speech_args[SPEECH_MESSAGE])
+	if(owner && (!newmsg || newmsg == FALSE || newmsg == ""))
+		to_chat(owner, span_warning("An implant prevents you from speaking!"))
+		speech_args[SPEECH_MESSAGE] = ""
+		return
+	speech_args[SPEECH_MESSAGE] = newmsg
+
+
+
+/obj/item/organ/cyberimp/brain/wordblocker/examine(mob/user)
+	. = ..()
+	. += "<span class='info'>A little screen reads: 'Implant Unique ID: [id]'</span>"
+	if(mgr)
+		. += "<span class='info'>The implant currently does [mgr.visible ? "" : "not"] allow remote access. Use a screwdriver to change this.</span>"
```

Comment:
```suggestion
		. += "<span class='info'>The implant currently [mgr.visible ? "allows" : "does not allow"] remote access. Use a screwdriver to change this.</span>"
```
laugh at the ESL part two

Reply:
Both are correct:
"[it] does allow remote access" and "[it] does not allow remote access"
"[it] allows remote access" and "[it] does not allow remote access"
the word "does" in the former adds emphasis to the verb "allow".

Reply:
*fixes sentence that was already correct*
"laugh at ESL"